File: 프로그래머스/lv2/42885. 구명보트/구명보트.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> people, int limit) {
    sort(people.begin(), people.end());
    
    int left = 0, right = people.size() - 1;
    int answer = 0;
    
    while(left <= right) {
        if(left == right) { // last 1
            answer++;
            break;
        }
        
        if(people[left] + people[right] > limit) { // only right
            answer++;
            right--;
        } else { // both
            answer++;
            left++;
            right--;
        }
    }
    return answer;
}

File: 프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    answer = 0
    
    people.sort()
    lo, hi = 0, len(people) - 1
    while lo <= hi:
        if people[lo] + people[hi] <= limit:
            lo += 1
        
        hi -= 1
        answer += 1

    return answer